### PR TITLE
Ready for review

### PR DESCRIPTION
**Pull request by truealisa**

### CODE REVIEW

File: lib/rg_hw_codebreaker/cli.rb [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+module RgHwCodebreaker
+  # Class Cli responsible for console communication with player
+  class Cli
+    attr_reader :game
+
+    def initialize
+      @current_menu = :main_menu
+      @exit_break = false
+      @best_results = YAML.load_file(File.join(__dir__, 'results.yml'))
```
Comment (valikos) [inline lib/rg_hw_codebreaker/cli.rb]: I think there will be better create new or read existing file.

File: lib/rg_hw_codebreaker/results.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (valikos) [inline lib/rg_hw_codebreaker/results.yml]: I think would be better add this file to the `.gitignore`.

File: lib/rg_hw_codebreaker/cli.rb (hunk 1 above)
Comment (truealisa) [reply lib/rg_hw_codebreaker/cli.rb]: I've extracted loading results file logic to #load_results_file method (which checks existence of file and creates a new one if needed) and removed @best_results instance variable. Is it better?

File: lib/rg_hw_codebreaker/results.yml (hunk 2 above)
Comment (truealisa) [reply lib/rg_hw_codebreaker/results.yml]: Done

File: lib/rg_hw_codebreaker/cli_constants.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+module RgHwCodebreaker
+  class Cli
```
Comment (valikos) [inline lib/rg_hw_codebreaker/cli_constants.rb]: To be honest I don't like this file. 
First reason it is name. File name is `cli_constant` and we just describe few constants for `Cli` class. 
It it interesting idea, store constants in separate file with suffix `_constants`.
I think we can leave and replace to `cli.rb` only `MENU` constant. Other constants put in some `yaml` file which will be used like translations. 
I am not sure that we need all of this `_MSG` constants.

**Comment by valikos** [comment]
Looks good. I put one comment to a `cli_constants.rb` file.
It is up to you.

File: lib/rg_hw_codebreaker/cli_constants.rb (hunk 3 above)
Comment (truealisa) [reply lib/rg_hw_codebreaker/cli_constants.rb]: Ok. I just saw that this looong strings look ugly in the main ```cli.rb``` file, but couldn't think of better way to store them. Now I've followed your advice and replaced ```cli_constants.rb``` with ```cli_messages.yml``` and loading it to @messages at #initialize.

Thank you for review and comments!:)

File: lib/rg_hw_codebreaker/cli.rb [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+      end
+    end
+
+    def save_result?
+      puts 'Do you want to save your result?(y/n)'
+      save = Readline.readline('>>> ')
+      save == 'y'
+    end
+
+    def write_result_to_file(current_result)
```
Comment (valikos) [inline lib/rg_hw_codebreaker/cli.rb]: As I wrote in rack app you need extract this method in separate class.

File: lib/rg_hw_codebreaker/cli.rb [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+      best_results = YAML.load(load_results_file)
+      best_results << current_result
+      File.open(File.join(__dir__, 'results.yml'), 'w') do |file|
+        YAML.dump(best_results, file)
+      end
+    end
+
+    def load_results_file
```
Comment (valikos) [inline lib/rg_hw_codebreaker/cli.rb]: As I wrote in rack app you need extract this method in separate class.

**Comment by valikos** [comment]
Regarding `write` and `read` results. I thought that you detect that you are using methods from class which designed to terminal communication. In such situation `write` and `read` are common operations. So they should live separately 😄

**Comment by truealisa** [comment]
Fixed.
I had a thought that it is something that needs to be done, but then just forgot to change it...

File: lib/rg_hw_codebreaker/cli.rb (hunk 4 above)
Comment (truealisa) [reply lib/rg_hw_codebreaker/cli.rb]: Done

File: lib/rg_hw_codebreaker/cli.rb (hunk 5 above)
Comment (truealisa) [reply lib/rg_hw_codebreaker/cli.rb]: Done

**Pull request closed without merging**